Preorder Traversal in C

rohangawhade opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes
### Description

Added preorder traversal using recursion and iteration in C
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#415

rohangawhade committed:
Changed the location of preordertraversal.c from graphs to ds folder

rohangawhade committed:
Merge https://github.com/TesseractCoding/NeoAlgo

rohangawhade committed:
Added Preorder traversal in C

rohangawhade committed:
Update README.md

siddharth25pandey commented on C/ds/Preordertraversal.c:
```diff
@@ -0,0 +1,100 @@
+void preorderRecursive(struct BST*);
+void preorderIterative(struct BST*);
+
+int main(){
```
Display message before & after of  inserting element in the node

rohangawhade replied:
Added display message.

siddharth25pandey commented on C/ds/Preordertraversal.c:
```diff
@@ -0,0 +1,100 @@
+		    printf("%d ", root->data);
+			root=root->right;
+        }
+    }
```
Add Sample Input & Output

rohangawhade replied:
Added sample input and output.

rohangawhade committed:
Update Preordertraversal.c

Added display message after inserting element and added a sample output at the end of the program

atarax665 commented:
@all-contributors please add @rohangawhade  for code.
